order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java

```diff
@@ -70,10 +86,161 @@
 			return new ResponseEntity<>(new ApiErrorsWS(errorWSList), HttpStatus.UNPROCESSABLE_ENTITY);
 		}
 
-//		OrderBean orderBean = orderMapper.toBean(orderWS);
-//		OrderBean order = orderService.createOrder(orderBean);
+		OrderBean orderBean = orderMapper.toBean(orderWS, new CycleAvoidingMappingContext());
+
+		OrderBean order = orderService.createOrder(orderBean);
+
+		return new ResponseEntity<>(orderMapper.toWS(order, new CycleAvoidingMappingContext()), HttpStatus.OK);
+	}
+
+
+	@PutMapping("/order/{uuid}")
+	@RequiresJwtAuthentication
+	@RequiresPermissions("order:update")
+	public ResponseEntity<?> updateOrder(@PathVariable("uuid") String orderUuid, @Validated @RequestBody OrderWS orderWS) {
+
+		Optional<OrderBean> tmpOrderBean = orderService.findByUuid(orderUuid);
```

Comment:
You can prob create a orderService.orderExist(id), the repository can use exist query. Usually it's faster then retrieving the whole entity(won't be an issue here, but can avoid creating a tmpOrderBean).

Reply:
do you mean  orderService.orderExist(uuid)?

Reply:
fixed